pytest-embedded-idf/pytest_embedded_idf/dut.py

```diff
@@ -35,3 +48,31 @@ def toolchain_prefix(self) -> str:
             return f'riscv32-{self.target}-elf-'
         else:
             raise ValueError(f'Unknown target: {self.target}')
+
+    def image_info(self, path_to_file):
```

Comment:
Should this be a method of `App` rather than the `Dut`?
Looks like this method doesn't use anything from `IdfDut`, and is not dependent on even having a Dut instance?

Reply:
It uses `target` from `IdfDut`

Reply:
IdfApp also has the .target property, right?

I'm just thinking that getting the SHA256 hash attached to the binary image seems to be more like a utility function related to the app binary image. It doesn't look related to the DUT. Besides, IdfApp also "knows" the app binary path, so the benefit is that it doesn't need to be passed as an argument.

Reply:
I think this is okay to move to IdfApp. but since IdfApp is not 100% guaranteed includes esptool when using, (for example it is used alone without service `esp`), I recommend the `import esptool` to be put inside the function itself.

Reply:
and we don't need a try except ImportError by this approach.